nexus_constructor/transformation.py

```diff
@@ -0,0 +1,63 @@
+import h5py
+from nexus_constructor.vector import Vector
+from abc import ABC, abstractmethod
+
+
+class Transformation(ABC):
+    """
+    Parent class for transformations on the model.
+    """
+
+    @abstractmethod
+    def __init__(self, name):
+        """
+        Creates an in-memory nexus file with a named entry containing an NXTransformation attribute.
+        :param name: The name of the root entry in the nexus file.
+        """
+        self.nexus_file = h5py.File(name, driver="core", backing_store=False)
+        self.transformation = self.nexus_file.create_group(name)
+        self.transformation.attrs["NX_class"] = "NXtransformation"
+
+    @property
+    def name(self):
+        return self.transformation.name[1:]
```

Comment:
Why do you drop the first character?

Reply:
It returns the full path which contains forward slashes.